go/vt/vtgate/planbuilder/insert.go

```diff
@@ -133,8 +207,8 @@ func findOrInsertPos(ins *sqlparser.Insert, col cistring.CIString, rowNum int) (
 		pos = len(ins.Columns)
 		ins.Columns = append(ins.Columns, sqlparser.ColIdent(col))
 	}
-	if pos == -1 || pos >= len(ins.Rows.(sqlparser.Values)[rowNum].(sqlparser.ValTuple)) {
-		ins.Rows.(sqlparser.Values)[rowNum] = append(ins.Rows.(sqlparser.Values)[rowNum].(sqlparser.ValTuple), &sqlparser.NullVal{})
+	if pos >= len(ins.Rows.(sqlparser.Values)[rowNum]) {
```

Comment:
It's impossible for pos to be -1 here. The check was unnecessary and confusing. So, I removed it.